src/main/java/us/kbase/auth/client/AuthClient.java

```diff
@@ -135,5 +153,30 @@ public String getServerVersion() throws IOException, AuthException {
 		final Map<String, Object> doc = request(rootURI);
 		return (String) doc.get("version");
 	}
+	
+	// TODO CODE could do a lot more here later w/ the return data from the auth server
+	//           - token type, custom expiration time, etc.
+	
+	/** Validate a token and get name of the user that owns the token.
+	 * @param token the token.
+	 * @return an authtoken containing the token and the username.
+	 * @throws IOException if an IOException occurs communicating with the auth service.
+	 * @throws AuthException if an auth exception occurs communicating with the auth service.
+	 */
+	public AuthToken validateToken(final String token) throws IOException, AuthException {
+		if (token == null || token.trim().isEmpty()) {
+			throw new IllegalArgumentException("token must be a non-whitespace string");
+		}
+		final AuthToken t = tokenCache.getToken(token);
+		if (t != null) {
+			return t;
+		}
+		final URI target = rootURI.resolve("api/V2/token");
```

Comment:
do we need to setRequestMethod here when calling target url? 
api/v2/token could be either GET or POST based on auth2 doc

Reply:
👍